packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -778,7 +778,12 @@ class AndroidDevice extends Device {
 
   @override
   void clearLogs() {
-    _processUtils.runSync(adbCommandForDevice(<String>['logcat', '-c']));
+     final RunResult result = _processUtils.runSync(adbCommandForDevice(<String>['logcat', '-c']));
+     // Do not log to standard error because that causes test to fail.
+     if (result.exitCode != 0) {
+      _logger.printTrace('"adb logcat -c" failed: exitCode: ${result.exitCode}'
+        ' stdout: ${result.stdout} stderr: ${result.stderr}');
+    }
```

Comment:
I'm afraid I've sent you down the wrong path; after digging through PorcessUtils; it shouldn't fail.  `RunResult` is basically after the fact.   The "Task failed:" string in the logs only shows up here:

./dev/devicelab/lib/framework/framework.dart:280:      final String message = 'Task failed: $taskError';

The "Executable" part of the string looks like its from:
./dev/devicelab/lib/framework/utils.dart:383:    fail('Executable "$executable" failed with exit code $exitCode.');
 
if I change the way I search; there is another logcat: 

https://github.com/flutter/flutter/blob/034966b3ccc4a362fd25f269f2c6850354359da2/dev/devicelab/lib/framework/devices.dart#L736-L739

```dart
  Future<void> startLoggingToSink(IOSink sink, {bool clear = true}) async {
    if (clear) {
      await adb(<String>['logcat', '--clear'], silent: true);
    }
```

`adb` here calls `eval` in utils with `canFail=false`. I believe this is the route taken to fail the test and we can just use `canFail=true`.

Thoughts?